ykpack/src/types.rs

```diff
@@ -93,7 +96,16 @@ impl Display for BasicBlock {
 pub enum Statement {
     Nop,
     Assign(Place, Rvalue),
-    Unimplemented, // FIXME
+    SetDiscriminant{place: Place, variant_index: VariantIndex},
```

Comment:
To fit in with the other members of the `enum`, should this be `SetDiscriminant(Place, VariantIndex)`?

Reply:
Yeah, I've been pondering on this myself. In MIR it's a random mix.

Ideally I'd use only tuple-like variants, but I know I'd be upset that I'd have to break my rule if we had a variant with lots of the same type.

E.g.:
```
enum Taps {
    ...
    Mixer(usize, usize, usize, usize),
}
```

Here a struct-like would be preferred, so that we can name the fields.

Thoughts?

Reply:
I agree that when you get repetitions of the same type you want to use names, but that's not the case here. [I also sometimes use names when the type doesn't give much away about what the "thing" is.] Up to you.

Reply:
OK